java/statestore/src/main/java/sleeper/statestore/dynamodb/DynamoDBStateStore.java

```diff
@@ -37,15 +37,21 @@ public class DynamoDBStateStore extends DelegatingStateStore {
     public static final String PARTITION_ID = DynamoDBPartitionFormat.ID;
     public static final String PARTITION_ID_AND_FILENAME = DynamoDBFileInfoFormat.PARTITION_ID_AND_FILENAME;
     public static final String TABLE_ID = "TableId";
+    public static final int DEFAULT_PAGE_LIMIT = 1024 * 1024;
 
     public DynamoDBStateStore(InstanceProperties instanceProperties, TableProperties tableProperties, AmazonDynamoDB dynamoDB) {
+        this(instanceProperties, tableProperties, dynamoDB, DEFAULT_PAGE_LIMIT);
+    }
+
+    public DynamoDBStateStore(InstanceProperties instanceProperties, TableProperties tableProperties, AmazonDynamoDB dynamoDB, int pageLimit) {
         super(DynamoDBFileInfoStore.builder()
                         .dynamoDB(dynamoDB)
                         .activeTableName(instanceProperties.get(ACTIVE_FILEINFO_TABLENAME))
                         .readyForGCTableName(instanceProperties.get(READY_FOR_GC_FILEINFO_TABLENAME))
                         .sleeperTableId(tableProperties.get(TableProperty.TABLE_ID))
                         .stronglyConsistentReads(tableProperties.getBoolean(DYNAMODB_STRONGLY_CONSISTENT_READS))
                         .garbageCollectorDelayBeforeDeletionInMinutes(tableProperties.getInt(GARBAGE_COLLECTOR_DELAY_BEFORE_DELETION))
+                        .pageLimit(pageLimit)
```

Comment:
It's a bit confusing that this page limit field is actually only used in the query to list active files.

Maybe it would be worth building the DynamoDBFileInfoStore directly in the tests that need to set this, so that we don't have to worry about it when constructing the whole state store? We could also make sure it's used in all queries that don't otherwise set a limit?